### DISCUSSION TITLE
debian: how to build from source?

### DISCUSSION THREAD

**Discussion by sid-the-sloth** [lah7/gtk3-classic#69] [Help & Support] [answered]
Could somebody please write a couple of lines about how to build this from source on Debian? Thanks!

**Comment by lah7** [comment]
There's an _"easy"_ method, but it's the messiest to uninstall and causes two versions to co-exist (this installs to `/usr/local`, which overrides anything installed in `/usr`, like the system's original GTK3)

1. Install the build dependencies:

       sudo apt-get build-dep gtk+3.0

2. [Download GTK 3's source and a copy of this repository; then apply our patches](https://github.com/lah7/gtk3-classic/wiki/Creating-and-Managing-Patches).

3. Check out [Arch's PKGBUILD to see the build commands](https://github.com/lah7/gtk3-classic/blob/master/PKGBUILD#L115-L144). They use Ninja/Meson for the build system, the question becomes "how to build gtk3?"

**Comment by lah7** [comment, marked as answer]
An alternate method is to patch the  `gtk+3.0` source package and adding our patches (removing any existing ones that conflict). This is how it's done for Ubuntu, which is [automated by this script](https://github.com/lah7/gtk3-classic/blob/master/scripts/create-ubuntu-package.sh).  It's cleaner since you can re-install the original GTK+3 to revert back.

First, enable the source code repositories in Apt and downloading the package source:

    apt-get source gtk+3.0

[Download a copy of this repository](https://github.com/lah7/gtk3-classic/archive/refs/heads/master.zip), add our `*.patch` files into `debian/patches` and update the `series` file. Then, [here's how to build a package](https://www.debian.org/doc/manuals/maint-guide/build.en.html). Not sure on the exact command.

#### Some gotchas:

* Debian's packaging system is **really picky** about fuzzy quilt patches, so they might need individually refreshing first.
* The package might not build locally without workarounds - at least, this is the case with Ubuntu's `gtk+3.0` (based on Debian's). There might be a file somewhere that mentions this.
* Make sure to update `debian/rules` and [remove the checks](https://github.com/lah7/gtk3-classic/blob/master/scripts/create-ubuntu-package.sh#L91), since the classic patches will cause it to fail,

**Reply by sid-the-sloth** [reply]
Thanks! I'll stick to `gtk3-nocsd` until this makes it into XFCE source. Thank you for doing this, and I do hope you can convince the XFCE team to include something like this in the source so anyone can keep the clean classic look of their desktop.
